File: src/apps/jhonny_home_studio_admin/src/app/core/utils/duration-formatter.ts
export class DurationFormatter {
  private constructor() {}

  static format(minutes: number | null | undefined): string {
    const normalizedMinutes = Number(minutes ?? 0);

    if (!Number.isFinite(normalizedMinutes) || normalizedMinutes <= 0) {
      return 'Tempo a confirmar';
    }

    if (normalizedMinutes < 60) {
      return normalizedMinutes === 1 ? '1 minuto' : `${normalizedMinutes} minutos`;
    }

    const hours = Math.floor(normalizedMinutes / 60);
    const remainingMinutes = normalizedMinutes % 60;

    if (remainingMinutes === 0) {
      return hours === 1 ? '1 hora' : `${hours} horas`;
    }

    return `${hours}h${remainingMinutes.toString().padStart(2, '0')}`;
  }

  static estimated(minutes: number | null | undefined): string {
    const normalizedMinutes = Number(minutes ?? 0);

    if (!Number.isFinite(normalizedMinutes) || normalizedMinutes <= 0) {
      return 'Tempo a confirmar';
    }

    return `Estimativa de ${this.format(normalizedMinutes)}`;
  }
}

File: src/apps/jhonny_home_studio_app/lib/core/utils/duration_formatter.dart
import 'service_presentation_formatter.dart';

class DurationFormatter {
  const DurationFormatter._();

  static String format(int minutes) {
    if (minutes <= 0) {
      return 'Tempo a confirmar';
    }

    if (minutes < 60) {
      return minutes == 1 ? '1 minuto' : '$minutes minutos';
    }

    final hours = minutes ~/ 60;
    final remainingMinutes = minutes % 60;

    if (remainingMinutes == 0) {
      return hours == 1 ? '1 hora' : '$hours horas';
    }

    return '${hours}h${remainingMinutes.toString().padLeft(2, '0')}';
  }

  static String estimated(int minutes) {
    return ServicePresentationFormatter.estimatedDuration(minutes);
  }
}
